civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -1723,6 +1731,32 @@ function civihr_employee_portal_menu() {
   return $items;
 }
 
+/**
+ * Callback function for 'emergency_contacts/{id}/delete
+ *
+ * @param string $id
+ *   The emergency contact ID
+ */
+function civihr_employee_portal_emergency_contact_delete($id) {
+  $service = new EmergencyContactDeletionNotifier();
+  $result = ['status' => 200];
+
+  try {
+    civicrm_api3('Contact', 'deleteemergencycontact', ['id' => $id]);
+    $service->notifyDeletion($id);
+  } catch (\Exception $exception) {
+    $result = [
+      'status' => 500,
+      'error' => $exception->getMessage(),
+    ];
+  }
+
+  drupal_add_http_header('Content-Type', 'application/json');
+  http_response_code($result['status']);
+  echo json_encode($result);
+  exit();
```

Comment:
I think we don't need to do anything about the response here, but if that is really necessary, the right way to do that in drupal is by setting the **delivery callback** in hook_menu to `drupal_json_output`: https://drupal.stackexchange.com/questions/149157/how-to-return-jsonp-from-a-menu-callback

Reply:
I was using this, but it did some strange things like encoding quotes differently so I thought I'd do it myself. If you think we don't need to have any special response (see comment above) then I'll remove all output.